### PR TITLE
config/validate: add some sanity checks

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by crawford** [comment]
/cc @marineam for checkWriteFiles() and @xiang90 for checkDiscoveryUrl()

Are these checks accurate/exhaustive?

File: config/validate/rules.go [hunk 1]
```diff
@@ -29,6 +32,9 @@ type rule func(config node, report *Report)
 var Rules []rule = []rule{
 	checkStructure,
 	checkValidity,
+	checkWriteFiles,
+	checkDiscoveryUrl,
```
Comment (jonboulle) [inline config/validate/rules.go]: Is the ordering significant?

File: config/validate/rules.go [hunk 2]
```diff
@@ -98,6 +104,54 @@ func checkNodeValidity(n, g node, r *Report) {
...
+	if !c.IsValid() {
+		return
+	}
+
+	if _, err := url.ParseRequestURI(c.String()); err != nil {
+		report.Warning(c.line, "discovery URL is not valid")
+	}
+}
+
+func checkWriteFilesUnderCoreos(cfg node, report *Report) {
```
Comment (jonboulle) [inline config/validate/rules.go]: You forgot to wire this one up

File: config/validate/rules.go (hunk 1 above)
Comment (crawford) [inline config/validate/rules.go]: Nope. All rules are evaluated.

File: config/validate/rules.go (hunk 2 above)
Comment (crawford) [inline config/validate/rules.go]: oops. fixed.

File: config/validate/rules.go (hunk 1 above)
Comment (jonboulle) [inline config/validate/rules.go]: imho just sort lexicographically then, it makes it slightly clearer there is no other significance

Comment (crawford) [inline config/validate/rules.go]: good point

**Comment by crawford** [comment]
@jonboulle learned my ABCs

**Comment by jonboulle** [comment]
lgtm

**Pull request merged**